sql/sql_class.h

```diff
@@ -3760,6 +3760,11 @@ class THD :public Statement,
       killed_err= 0;
       mysql_mutex_unlock(&LOCK_thd_kill);
     }
+#ifdef WITH_WSREP
+    mysql_mutex_lock(&LOCK_thd_data);
+    wsrep_aborter= 0;
+    mysql_mutex_unlock(&LOCK_thd_data);
+#endif /* WITH_WSREP */
   }
```

Comment:
Is it really necessary to use a mutex here? This could introduce a performance regression that affects even the non-Galera use case.

Could we use `my_atomic_store32_explicit(&wsrep_aborter, 0, MY_MEMORY_ORDER_RELAXED)` and declare the data member as `int32`? (Before 10.4, the C++11 `std::atomic` or `std::atomic_flag` cannot be used.) And of course, this suggestion would imply that all use of the flag be covered by `my_atomic` and possibly additionally by a mutex.

Also, why are we using `LOCK_thd_data` here, while in `THD::awake()` the data member appears to be protected by `LOCK_thd_kill`?

Reply:
I will move this inside LOCK_thd_kill protection.